hibernate-core/src/main/java/org/hibernate/dialect/H2Dialect.java

```diff
@@ -210,8 +210,8 @@ public H2Dialect() {
 		registerFunction( "curtime", new NoArgSQLFunction( "curtime", StandardBasicTypes.TIME ) );
 		registerFunction( "curtimestamp", new NoArgSQLFunction( "curtimestamp", StandardBasicTypes.TIME ) );
 		registerFunction( "current_date", new NoArgSQLFunction( "current_date", StandardBasicTypes.DATE ) );
-		registerFunction( "current_time", new NoArgSQLFunction( "current_time", StandardBasicTypes.TIME ) );
-		registerFunction( "current_timestamp", new NoArgSQLFunction( "current_timestamp", StandardBasicTypes.TIMESTAMP ) );
+		registerFunction( "current_time", new NoArgSQLFunction( buildId >= 200 ? "localtime" : "current_time", StandardBasicTypes.TIME ) );
```

Comment:
Shouldn't this (and the next line) be >=198 instead?
https://github.com/hibernate/hibernate-orm/pull/3412#issuecomment-678813877

My sentence
> As of 1.4.200 other functions also include the time zone and might fail as well but those changes might have been made after 1.4.198.
might have been a bit misleading. "As of 1.4.200" means that the current online documentation (which matches 1.4.200) is saying that even more functions include the time zone but I haven't checked whether those functions were added also in 198 or in 199 or 200.

Reply:
Well I know for sure that the both forms work fine on 1.4.199 because I've tested them both in H6.

It's a bit harder to test them on H5 because there are other distracting test failures.

Reply:
I mean the real problem here is that he's insisting on not implicitly casting `TIME WITH TIME ZONE` to `TIME`, a cast which I'm pretty sure is implicit on most other databases.